scripts/spack/packages/axom/package.py

```diff
@@ -69,6 +69,8 @@ class Axom(CachedCMakePackage, CudaPackage):
     variant("hdf5",     default=True, description="Build with hdf5")
     variant("lua",      default=True, description="Build with Lua")
     variant("scr",      default=False, description="Build with SCR")
+    variant("kvtree",   default=False, description="Build with kvtree")
+    variant("dtcmp",    default=False, description="Build with dtcmp")
     variant("umpire",   default=True, description="Build with umpire")
```

Comment:
Do we want the `kvtree` `dtcmp` variants?
It seems that we will always need these two when `+scr`  (unless I'm missing something)

Reply:
I think this will  also need the following update below:

```py
depends_on("scr", when="+scr")
depends_on("kvtree", when="+scr")
depends_on("ctcmp", when="+scr")
```

Perhaps they can all be combined into one line?

Tag: @white238

Reply:
I'm assuming the axom dependency is needed so we can add their paths to our host-config.

It would be even better if `scr`'s cmake export handled this for us, so axom's build system does not need to be aware of `kvtree` or `dtcmp`.
(I don't know enough details about these, but it didn't look like axom actually needed these -- only scr).

Reply:
I would remove the kvtree and dtcmp variants since we would never want a customer to turn them off.  The variant is only if we have scr support, which then makes us rely on scr, kvtree, and dtcmp

Reply:
If I don't have the variants, then the the ${KVTREE_DIR} and ${DTCMP_DIR} variables that are needed in FindSCR.cmake are not created and those libraries can't be found for the axom build.  It doesn't seem that SCR creates the cmake export files, at least not in a way that I could figure out.  It took me a while to figure out something that works here.

Reply:
That's because you need to remove `scr`, `kvtree`, `dtcmp` from line 306 and give them its own section, like so:


```
       # Handle all of scr's deps
        if '+scr' in spec:
            for dep in ('scr', 'kvtree', 'dtcmp',):
                dep_dir = get_spec_path(spec, dep, path_replacements)
                entries.append(cmake_cache_path('%s_DIR' % dep.upper(),
                                                dep_dir))
       else:
           entries.append('# scr not build\n')
```

Reply:
OK, I made this change.  I should be able to merge if the CI passes.